packages/terra-dev-site/src/webpack/plugin/SitePlugin.js

```diff
@@ -341,6 +359,72 @@ class SitePlugin {
       oneTimeSetupComplete = false;
       siteRegistry = {};
     });
+
+    // Evaluate all URLS found in the site for correctness.
+    compiler.hooks.done.tap('terra-dev-site-plugin', () => {
+      // Convert generated links for all sites into a single list.
+      const validRelativeUrls = Object.values(generatedBootstrapData).reduce((acc, urls) => acc.concat(urls), []);
+      // add the favicon since it's valid too.
+      validRelativeUrls.push(`${publicPath.slice(0, -1)}/favicon.ico`);
+
+      // Log formatting
+      const invalidUrlLog = (url) => `${chalk.blue(url.url)} in:\n ${chalk.yellow(url.file.history[0].replace(url.file.cwd, ''))} at line ${url.node.position.start.line}`;
+
+      // Create a promise array to allow all logs to happen at the same time.
+      const resultPromises = [];
+      const queueLength = urlQueue.length;
+      for (let index = 0; index < queueLength; index += 1) {
+        const url = urlQueue.shift();
+
+        if (url.url.startsWith('/')) {
+          // for relative urls check against the know list of valid relative urls.
+          // Remove query params or hash links from the end of the url.
+          const urlWithoutHash = url.url.replace(/[#?].*$/, '');
+          if (!validRelativeUrls.includes(urlWithoutHash)) {
+            resultPromises.push(Promise.resolve(`${chalk.red('[Broken Link]')} ${invalidUrlLog(url)}`));
+          }
+        } else if (url.url.startsWith('https')) {
+          // required to not assume https urls are invalid.
```

Comment:
so we could validate more link types, http, https and even maybe ./ style relative links, but there would need to be some more changes. 

For http and https style links, the real trouble is that link to engineering.cerner.com/terra-ui always come back as 404s and we don't currently have a way to check if the links are valid. That the links come back as 404s is not unexpected because we use a custom 404 to enable the browser router and to redirect to the correct page. We would need to look into further identification of what is a bad link in a deployed site for this to work out.

for ./ style relative links we'd need to generate the site path based on the file that is hosting the link. Doable but im not sure it's worth it.